diff --git a/fedimint-clientd/src/router/handlers/ln/mod.rs b/fedimint-clientd/src/router/handlers/ln/mod.rs
--- a/fedimint-clientd/src/router/handlers/ln/mod.rs
+++ b/fedimint-clientd/src/router/handlers/ln/mod.rs
@@ -16,7 +16,8 @@
 pub mod invoice_external_pubkey_tweaked;
 pub mod list_gateways;
 pub mod pay;
 pub mod readonly_quote;
+pub mod connect_federation;
 
 pub async fn get_invoice(req: &LnPayRequest) -> anyhow::Result<Bolt11Invoice> {
     let info = req.payment_info.trim();
diff --git a/fedimint-clientd/src/main.rs b/fedimint-clientd/src/main.rs
--- a/fedimint-clientd/src/main.rs
+++ b/fedimint-clientd/src/main.rs
@@ -275,6 +275,8 @@
         .route("/pay", post(ln::pay::handle_rest))
         .route("/list-gateways", post(ln::list_gateways::handle_rest))
-        .route("/ecashmesh-quote", post(ln::readonly_quote::handle_rest));
+        .route("/ecashmesh-quote", post(ln::readonly_quote::handle_rest))
+        .route("/ecashmesh-connect-preview", post(ln::connect_federation::connect_preview))
+        .route("/ecashmesh-connect", post(ln::connect_federation::connect_federation));
 
     let onchain_router = Router::new()
         .route(
